# Introduction

With this data set, we will examine the gold export structure of Ghana and Nigeria worldwide. The aim is to evaluate the gold export trends, volumes and impact of these two countries on the world market. Dataset gathered form UN Comtrade.

## Dataset Columns & Definitions

**TypeCode**: Identifies the type of record, typically indicating whether it's commercial goods, services, etc.

**FreqCode**: Indicates the frequency of the data report (e.g., 'M' for monthly, 'A' for annual).

**RefPeriodId**: A unique identifier for the reference period of the data, often combining year and month.

**RefYear**: The year to which the data pertains.

**RefMonth**: The month to which the data pertains.

**Period**: A combined year and month format that specifies the exact period of the data report.

**ReporterCode**: Numeric code representing the country that reported the data.

**ReporterISO**: The ISO code of the reporting country.

**ReporterDesc**: Descriptive name of the reporting country.

**FlowCode**: Code indicating the direction of trade flow (e.g., exports or imports).

**FlowDesc**: Description of the trade flow type.

**PartnerCode**: Numeric code representing the partner country in the trade.

**PartnerISO**: The ISO code of the partner country.

**PartnerDesc**: Descriptive name of the partner country.

**Partner2Code**: Numeric code for a secondary partner country, if applicable.

**Partner2ISO**: ISO code of a secondary partner country.

**Partner2Desc**: Description of a secondary partner country.

**ClassificationCode**: Code of the classification used for the products or services in the report.

**ClassificationSearchCode**: A searchable version of the classification code.

**IsOriginalClassification**: Boolean indicating whether the classification is the original one used by the reporting country.

**CmdCode**: Code for the commodity or service reported.

**CmdDesc**: Description of the commodity or service.

**AggrLevel**: Level of aggregation of the data.

**IsLeaf**: Boolean indicating if the data point is at the lowest level of the classification tree.

**CustomsCode**: Code used by customs for the reported items.

**CustomsDesc**: Description of the customs code.

**MosCode**: Code indicating the mode of shipment.

**MotCode**: Mode of transport code.

**MotDesc**: Description of the mode of transport.

**QtyUnitCode**: Code for the unit of quantity in which the goods are measured.

**QtyUnitAbbr**: Abbreviation of the quantity unit.

**Qty**: Quantity of the goods traded.

**IsQtyEstimated**: Boolean indicating whether the quantity is estimated.

**AltQtyUnitCode**: Code for an alternate unit of quantity.

**AltQtyUnitAbbr**: Abbreviation for the alternate unit of quantity.

**AltQty**: Alternate quantity measurement, if used.

**IsAltQtyEstimated**: Boolean indicating whether the alternate quantity is estimated.

**NetWgt**: Net weight of the goods traded.

**IsNetWgtEstimated**: Boolean indicating whether the net weight is estimated.

**GrossWgt**: Gross weight of the goods traded.

**IsGrossWgtEstimated**: Boolean indicating whether the gross weight is estimated.

**Cifvalue**: CIF (Cost, Insurance, Freight) value of the goods imported.

**Fobvalue**: FOB (Free on Board) value of the goods exported.

**PrimaryValue**: The primary monetary value of the trade.

**LegacyEstimationFlag**: Indicates if the values are based on legacy estimation methods.

**IsReported**: Boolean indicating whether the data was directly reported by the country.

**IsAggregate**: Boolean indicating whether the data is an aggregation of multiple records.

## Libraries

In [11]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

### Data Import

In [15]:
# In local, I identify the current path of dataset
file_path = r"C:\Users\MRE\Desktop\Data Science & Business\Project\Dataset\aga_bu_son_olsun_artik.csv"

# Then I upload
df = pd.read_csv(file_path, encoding='iso-8859-1')

In [16]:
# First look to dataset
df.head()

TypeCode FreqCode  RefPeriodId  RefYear  RefMonth  Period  ReporterCode  \
0        C        A     20170101     2017        52    2017           288   
1        C        A     20170101     2017        52    2017           288   
2        C        A     20170101     2017        52    2017           288   
3        C        A     20170101     2017        52    2017           288   
4        C        A     20170101     2017        52    2017           288   

  ReporterISO ReporterDesc FlowCode  ... IsNetWgtEstimated  GrossWgt  \
0         GHA        Ghana        X  ...              True       NaN   
1         GHA        Ghana        X  ...             False       NaN   
2         GHA        Ghana        X  ...             False       NaN   
3         GHA        Ghana        X  ...             False       NaN   
4         GHA        Ghana        X  ...             False       NaN   

  IsGrossWgtEstimated Cifvalue      Fobvalue  PrimaryValue  \
0               False      NaN  5.858281e+09  5.858281e+09   
1               False      NaN  1.103010e+06  1.103010e+06   
2               False      NaN  9.274560e+05  9.274560e+05   
3               False      NaN  4.926127e+03  4.926127e+03   
4               False      NaN  1.940124e+04  1.940124e+04   

  LegacyEstimationFlag IsReported IsAggregate  Unnamed: 47  
0                    0      False        True          NaN  
1                    0      False        True          NaN  
2                    0      False        True          NaN  
3                    0      False        True          NaN  
4                    0      False        True          NaN  

[5 rows x 48 columns]

## Data Exploration

In [18]:
# Let's look the shape of dataset
df.shape

(368, 48)

##### As we can see here, dataset has 368 rows, and 48 columns.